core/mm/core_mmu.c

```diff
@@ -255,6 +259,47 @@ static struct tee_mmap_region *find_map_by_pa(unsigned long pa)
 	return NULL;
 }
 
+#if defined(CFG_SECURE_DATA_PATH)
+static bool dtb_get_sdp_region(void)
+{
+	void *fdt = NULL;
+	int node;
+	paddr_t tmp_addr = 0;
+	size_t tmp_size = 0;
+
+	if (!IS_ENABLED(CFG_EMBED_DTB))
+		return false;
+
+	fdt = get_embedded_dt();
+	if (!fdt)
+		panic("No DTB found");
+
+	node = fdt_node_offset_by_compatible(fdt, -1, tz_sdp_match);
+	if (node < 0) {
+		DMSG("Cannot find %s node in the device tree", tz_sdp_match);
+		return false;
+	}
```

Comment:
Could you just add a test like the following:
```c
	while (node >= 0) {
		node = fdt_node_offset_by_compatible(fdt, node, tz_sdp_match);
		if (node >= 0)
			DMSG("Ingore SDP pool node %s, supports only 1 node",
			     fdt_get_name(fdt, node, NULL));
	}
```

Reply:
my fault: this implementation scratches `node` and line 289 will fail. Need another temp node reference:
```c
	int n = 0;
	(...)

	n = node;
	while ((n = fdt_node_offset_by_compatible(fdt, n, tz_sdp_match) >= 0)
		DMSG("Ignoring SDP pool node %s", fdt_get_name(fdt, n, NULL))

Reply:
indeed, I did not notice it either